### ISSUE TITLE
Support of non-exact FeedbackMessages in JQuery(Fenced)FeedbackPanel

### ISSUE THREAD

**Issue by Patrick1701** [sebfz1/wicket-jquery-ui#219]
Hi Sebastien,
do you think it would be possible to implement ```getIconClass(FeedbackMessage)``` and ```getMessageClass(FeedbackMessage)``` like wickets ```FeedbackMessage.isWarning()``` or e.g. ```FeedbackMessage.isError()```?

Wickets levels are working with ranges, so it is possible to sublass ```FeedbackMessage``` and -for instance- define 350 as warning level, as well.

Due to the exact-matching of  ```getIconClass(FeedbackMessage)``` and ```getMessageClass(FeedbackMessage)``` all messages are rendered without JQuery styling.

kind regards
Patrick

**Comment by sebfz1** [comment]
Hi Patrick,

Humm, it seems to be an inconsistency in `FeedbackPanel`...

You are right regarding the `#isLevel` method because it performs a "equals or greater than" operation, but the `FeedbackPanel#getCSSClass` uses a switch statement [1].

I guess the first statement should be the correct one, but it would be nice to confirm... Could you please ask on dev@? You might probably open a JIRA afterward, to ask updating `#getCSSClass` accordingly... ;) Then I will update `JQueryFeedbackPanel` at the same time...

[1] https://github.com/apache/wicket/blob/wicket-7.x/wicket-core/src/main/java/org/apache/wicket/markup/html/panel/FeedbackPanel.java#L294

**Comment by Patrick1701** [comment]
Oh, I didn't see that Wicket does it the same way internally...
Hmm... when I think about it again, maybe its not a good idea to use wickets range of message levels.
It's out of my context...

Now, I implemented my FeedbackMessage subclass in the way, it just uses the static warning level of Wickets ```FeedbackMessage```. So, I wont get problems, when something changes in the future.

But my origin problem is still there...
What I try to achieve is, implementing a seperate ```JQueryFencecFeedbackPanel``` which displays special messages. And I used the ```ExactLevelFeedbackMessageFilter``` to do that. Now, I just have a own impl. by checking for ```instanceof``` my FeedbackMessage subclass.
But its not working anyway... so my core problem is another.

I will sleep an night over, and possibly will close this issue tomorrow.

Thanx for your help
Patrick
